apps/verticalhome/js/app.js

```diff
@@ -219,6 +221,12 @@
           window.removeEventListener('gaiagrid-layout-ready', this);
           break;
 
+        case 'visibilitychange':
+          setTimeout(function() {
+            document.body.style.overflow = document.hidden ? 'hidden' : '';
```

Comment:
heh, nice :) I've been looking at similar hacks to this elsewhere too... It's a shame, because it indicates that our progressive tiled rendering code might not be working (and I think the cancelling code definitely isn't), but even if it was, it will make Gecko more responsive.